app/locations/controllers.js

```diff
@@ -4,39 +4,37 @@ async function locations(req, res, next) {
   const userPermissions = get(req.session, 'user.permissions', [])
 
   let regions = []
-
-  if (userPermissions.includes('allocation:create')) {
-    try {
-      regions = await req.services.referenceData.getRegions()
-    } catch (error) {
-      return next(error)
-    }
-  }
-
   let userLocations = req.userLocations
-
   const supplierId = req.session.user.supplierId
 
-  if (supplierId) {
-    userLocations = await req.services.referenceData.getLocationsBySupplierId(
-      supplierId
-    )
-  }
+  try {
+    if (userPermissions.includes('allocation:create')) {
+      regions = await req.services.referenceData.getRegions()
+    }
 
-  if (userPermissions.includes('locations:contract_delivery_manager')) {
-    const suppliers = await req.services.referenceData.getSuppliers()
-    const supplierLocations = await Promise.all(
-      suppliers.map(async supplier => {
-        const locations = await req.services.referenceData.getLocationsBySupplierId(
-          supplier.id
-        )
-        return locations
-      })
-    )
+    if (supplierId) {
+      userLocations = await req.services.referenceData.getLocationsBySupplierId(
+        supplierId
+      )
+    }
 
-    // The locations have been uniqued based on title to prevent
-    // duplicates when multiple suppliers have the same location
-    userLocations = uniqBy(supplierLocations.flat(), 'id')
+    if (userPermissions.includes('locations:contract_delivery_manager')) {
+      const suppliers = await req.services.referenceData.getSuppliers()
+      const supplierLocations = await Promise.all(
+        suppliers.map(async supplier => {
+          const locations = await req.services.referenceData.getLocationsBySupplierId(
+            supplier.id
+          )
+          return locations
+        })
+      )
+
+      // The locations have been uniqued based on title to prevent
+      // duplicates when multiple suppliers have the same location
+      userLocations = uniqBy(supplierLocations.flat(), 'id')
+    }
+  } catch (error) {
+    return next(error)
   }
 
   req.session.user.locations = userLocations
```

Comment:
It won't let me comment on the line but lines 42 to 46 were the ones removed by my PR so be good to just double check they don't creep back in with this. Will likely be resolved with a rebase :)

Reply:
Git was clever enough to not make a fuss about this, so all is as it should be :)